File: src/packets/PacketC2SProperty.js
const PacketC2S = require("./PacketC2S");
const Property = require("../Property");

/**
 * Client -> Server
 * ID = 0x34
 * Property
 */
class PacketC2SProperty extends PacketC2S {
    parseData() {
        this._property = this._buffer.readUInt8();
        switch(this._property) {
            case Property.PROPERTY_WEB_SERVER_ENABLED:
                break;
            case Property.PROPERTY_NAME:
                this._value = this._buffer.readStringNT();
                break;
        }
    }

    getProperty() {
        return this._property;
    }

    getValue() {
        return this._value;
    }

    getID() {
        return PacketC2SProperty.ID;
    }
}

PacketC2SProperty.ID = 0x34;

module.exports = PacketC2SProperty;

File: src/RNetPi.Core/Packets/PacketC2SProperty.cs
using RNetPi.Core.Constants;

namespace RNetPi.Core.Packets;

/// <summary>
/// Client -> Server
/// ID = 0x02
/// Property
/// </summary>
public class PacketC2SProperty : PacketC2S
{
    public const byte ID = 0x02;

    public byte Property { get; private set; }
    public object? Value { get; private set; }

    public PacketC2SProperty(byte[] data) : base(data)
    {
    }

    public override byte GetID() => ID;

    protected override void ParseData()
    {
        Property = Reader.ReadByte();
        switch (Property)
        {
            case Properties.WebServerEnabled:
                // No additional data
                break;
            case Properties.Name:
                Value = ReadNullTerminatedString();
                break;
            default:
                // Handle unknown properties gracefully
                break;
        }
    }
}